src/main/java/scala_maven/ScalaMojoSupport.java

```diff
@@ -514,7 +515,7 @@ private void addToClasspath(String groupId, String artifactId, String version, S
         addToClasspath(factory.createDependencyArtifact(d), classpath, addDependencies);
     }
 
-    void addToClasspath(Artifact artifact, Set<String> classpath, boolean addDependencies) throws Exception {
+    void addToClasspath(Artifact artifact, Collection<String> classpath, boolean addDependencies) throws Exception {
```

Comment:
The Set is to avoid duplicate (and need to remove duplicate)

Reply:
I'm happy to change the existing call sites use a `Set` if that's what you prefer, but the call sites currently use `List`. Classpath order (potentially) matters: If there are classes with the same name but different bytecode on the path then the order is relevant (I've not idea of the rest of maven really cares much about this TBH). So changing the call sites does introduce the possibility of subtle breakage for existing users. Your call.

Reply:
when the command send to scalac is sent, the plugin sort the classpath to have reproducible build (I opened the issue on scalac 2.7 .... I though it was solved). iirc the sort is done after the full classpath is defined (and before calling scalac....)